src/types/products.ts

```diff
@@ -0,0 +1,62 @@
+export interface IPrice {
+    current: number;
+    previous: number;
+    currency: string;
+}
+
+export interface IShipping {
+    type: string;
+    deliveryTime: string;
+}
+
+export interface IDetails {
+    freshness?: string;
+    farm?: string;
+    brand?: string;
+    model?: string;
+    delivery: string;
+    stock: string;
+}
+
+export interface IReview {
+    user: string;
+    rating: number;
+    comment: string;
+}
+
+export interface IQuestion {
+    user: string;
+    question: string;
+    answer: string;
+}
+
+export interface IExtraInfo {
+    descriptions: {
+        origins?: string;
+        howToCook?: string;
+        features?: string;
+        warranty?: string;
+    };
```

Comment:
Think it's better to have separate file for each type in this case

Reply:
Done.